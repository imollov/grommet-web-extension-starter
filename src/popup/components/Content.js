import React from 'react';
import browser from 'webextension-polyfill';
import { Box, Button, Text } from 'grommet';
import { Send } from 'grommet-icons';

const handleClick = () => browser.runtime.sendMessage({ popupMessage: 'hi' });

export default () => (
  <Box flex align="center" gap="large" pad="large">
    <Text>Powered by Grommet</Text>
    <Button
      label="Message Background"
      icon={<Send color="brand" />}
      onClick={handleClick}
    />
  </Box>
);
