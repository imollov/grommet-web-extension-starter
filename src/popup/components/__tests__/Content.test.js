import React from 'react';
import { shallow, mount } from 'enzyme';

import Content from '../Content';

describe('Content Component', () => {
  it('smoke test', () => {
    const wrapper = shallow(<Content />);

    expect(wrapper).toExist();
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Content />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call webextension API', () => {
    const wrapper = mount(<Content />);

    const link = wrapper.findWhere(
      el => el.text() === 'Message Background' && el.type() === 'button',
    );
    link.simulate('click');

    expect(global.browser.runtime.sendMessage).toHaveBeenCalled();
  });
});
