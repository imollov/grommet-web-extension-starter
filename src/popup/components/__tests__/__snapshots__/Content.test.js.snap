// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component should match snapshot 1`] = `
<Box
  align="center"
  flex={true}
  gap="large"
  pad="large"
>
  <Text
    level={1}
  >
    Powered by Grommet
  </Text>
  <Button
    icon={
      <Send
        color="brand"
      />
    }
    label="Message Background"
    onClick={[Function]}
  />
</Box>
`;
